lualib/lua_scanners/icap.lua

```diff
@@ -399,37 +403,47 @@ local function icap_check(task, content, digest, rule, maybe_part)
           http: HTTP/1.0 307 Temporary Redirect
         ]] --
 
+        local function string_to_threat_table(header_string)
```

Comment:
I would move this function outside of the enclosing function as it does not need to capture any external variables from it's scope.

Reply:
E.g. it can accept a `header_string` as a source and `threat_string` as a destination (wtf, it should be named as `threat_table` or a `threat_strings_list` for clarity!). In the closure it is totally unclear which side effects this function has. I would refrain from that.